automation/Extension/webext-instrumentation/src/background/dns-instrument.ts

```diff
@@ -0,0 +1,128 @@
+import { PendingResponse } from "../lib/pending-response";
+import { DnsResolved } from "../schema";
+import RequestFilter = browser.webRequest.RequestFilter;
+import ResourceType = browser.webRequest.ResourceType;
+import {
+  WebRequestOnCompletedEventDetails,
+} from "../types/browser-web-request-event-details";
+
+
+export class DnsInstrument {
+  private readonly dataReceiver;
+  private onCompleteListener;
+  private pendingResponses: {
+    [requestId: number]: PendingResponse;
+  } = {};
+  
+  constructor(dataReceiver) {
+    this.dataReceiver = dataReceiver;
+  }
+  
+  public run(crawlID){
+    // Adopted from http-instrument.ts
+    const allTypes: ResourceType[] = [
```

Comment:
Have you considered exporting this array in `http-instrument.ts` and importing it here, so that we only need to maintain it in one place?